vehicle/mkv/software/throttle/throttle.c

```diff
@@ -1,738 +1,276 @@
-		//         gCANMotorController[4],gCANMotorController[5],gCANMotorController[6],gCANMotorController[7]);
-		// LOG_println(msg2,strlen(msg2));
-
-	}
+/*
+ * Checks for plausibility with brake being pressed while throttle is also
+ * pressed.
+ *
+ * See 2022 Rules EV.5.7. When the brake is pressed and the throttle pedal is
+ * pressed beyond 25% pedal travel, we immediately set the torque request to
+ * zero and maintain that until throttle pedal is pressed less than 5%.
+ */
+static bool check_brake(int16_t pos_min) {
+    // Check brake plausibility (EV.5.7)
+    static bool brake_implausibility_occured = false;
+
+    if (throttle_state.brake_pressed) {
+        if (pos_min >= APPS_BRAKE_IMPLAUSIBILITY_THRESHOLD) {
+            brake_implausibility_occured = true;
+            gpio_set_pin(BRAKE_IMPLAUSIBILTIY_LED);
+            return true;
+        }
+
+        if (brake_implausibility_occured) {
+            if (pos_min <= APPS_BRAKE_IMPLAUSIBILITY_THRESHOLD_LOW) {
+                brake_implausibility_occured = false;
+                return false;
+            } else {
+                return true;
+            }
+        } else {
+            return false;
+        }
+    } else {
+        return false;
+    }
 }
 
-//******************Main Loop*****************
 int main(void) {
-	initTimer();
-	initADC();
-	sei();
-	CAN_init(CAN_ENABLED);
-	setPledOut();
-	LOG_init();
-	enablePCINT();
-	initMC();
-
-	CAN_wait_on_receive(MOB_DASHBOARD,
-	                    CAN_ID_DASHBOARD,
-	                    CAN_LEN_DASHBOARD,
-	                    0xFF);
-
-	CAN_wait_on_receive(MOB_BRAKELIGHT,
-	                    CAN_ID_BRAKE_LIGHT,
-	                    CAN_LEN_BRAKE_LIGHT,
-	                    0xFF);
-
-	CAN_wait_on_receive(MOB_PANIC,
-	                    CAN_ID_PANIC,
-	                    CAN_LEN_PANIC,
-	                    0xFF);
-
-	// PLED3_PORT |= _BV(PLED3);
-
-
-	while(1) {
-		if(bit_is_set(gTimerFlag,UPDATE_STATUS)) {
-			gTimerFlag &= ~_BV(UPDATE_STATUS);
-			storeThrottle();
-			getAverage();
-			checkPanic();
-			// checkShutdownState();
-			// testStartup();
-			printThrottle1();
-			sendCanMessages(0);
-
-			// gError = 5;
-			// showError();
-		}
-	}
+    can_init_throttle();
+    adc_init();
+    timer_init(&timer0_cfg);
+
+    gpio_set_mode(BRAKE_IMPLAUSIBILTIY_LED, OUTPUT);
+    gpio_set_mode(DEVIATION_IMPLAUSIBILITY_LED, OUTPUT);
+    gpio_set_mode(OUT_OF_RANGE_IMPLAUSIBILITY_LED, OUTPUT);
+    gpio_set_mode(RJ45_LED1, OUTPUT);
+    gpio_set_mode(RJ45_LED2, OUTPUT);
+
+    gpio_set_mode(SS_ESTOP, INPUT);
+    gpio_set_mode(SS_IS, INPUT);
+    gpio_set_mode(SS_BOTS, INPUT);
+
+    gpio_enable_interrupt(SS_ESTOP);
+    gpio_enable_interrupt(SS_IS);
+    gpio_enable_interrupt(SS_BOTS);
+
+    sei();
+
+    // Initialize motor controller direction
+    m192_command_message.direction_command = MOTOR_ANTICLOCKWISE;
+
+    can_send_m192_command_message();
```

Comment:
DOPE. This is super good to know, thanks Alex!

Reply:
Technically I never set it to 1, so we didn't have lockout, but also the inverter would never be enabled haha. I will set it so that as soon as we have ready-to-drive, we enable the inverter.